aries_vcx/src/messages/connection/service.rs

```diff
@@ -1,17 +1,46 @@
-use crate::messages::connection::did_doc::DidDoc;
+use crate::messages::connection::did_doc::Did;
+use crate::libindy::utils::ledger;
+use crate::error::prelude::*;
 
-#[derive(Debug, Deserialize, Serialize, Clone, PartialEq, Default)]
-#[serde(rename_all = "camelCase")]
-pub struct Service {
+pub const SERVICE_SUFFIX: &str = "indy";
+pub const SERVICE_TYPE: &str = "IndyAgent";
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(untagged)]
+pub enum ServiceResolvable {
+    FullService(FullService),
+    Did(Did)
+}
+
+impl ServiceResolvable {
+    pub fn resolve(&self) -> VcxResult<FullService> {
+        match self {
+            ServiceResolvable::FullService(full_service) => Ok(full_service.clone()),
+            ServiceResolvable::Did(did) => ledger::get_service(&did)
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+pub struct FullService {
+    pub id: Did,
+    #[serde(rename = "type")]
+    pub type_: String,
     #[serde(default)]
+    pub priority: u32,
+    #[serde(default)]
+    #[serde(rename = "recipientKeys")]
     pub recipient_keys: Vec<String>,
-    pub routing_keys: Option<Vec<String>>,
+    #[serde(default)]
+    #[serde(rename = "routingKeys")]
+    pub routing_keys: Vec<String>,
+    #[serde(rename = "serviceEndpoint")]
     pub service_endpoint: String,
 }
 
-impl Service {
+impl FullService {
```

Comment:
Since we renamed `Service` to `ServiceResolvable` (in the original version of the PR, I think we can keep here the original `Service` name

Reply:
Personally, I am torn on this one. I think that `Service` sounds way too general in a situation where there is a general enum representing a service, but possibly only a resolvable reference to it. There is an argument for calling the enum `Service` (since it is more general), and the struct `FullService` / `Service` as well. Hence, I would like to disambiguate between them very clearly by naming them very descriptively and differently.

Reply:
Idk I just don't see where is the `Full` coming from. The naming makes me think "okay this is `FullService`, so do we have some sort of 'incomplete'/'unfull' variant of service too?". It seems to me more accurate to just call the struct `Service`, since that's how tie field it reflects is called in DDO. 

Now that "wrapping enum" is called `ServiceResolvable `, whether you call this struct `FullService`, or `Service`, it's equally distinct from `ServiceResolvable` and I don't see an ambiguity in either case.

One more idea to throw in - what about calling it `DdoService`?

These are my thought's, I'll leave the final decision up to you, in the end this is mostly internal naming, so not too big of a deal.

Reply:
> Idk I just don't see where is the Full coming from. The naming makes me think "okay this is FullService, so do we have some sort of 'incomplete'/'unfull' variant of service too?". It seems to me more accurate to just call the struct Service, since that's how tie field it reflects is called in DDO.

I agree with that thinking, and to me, that "unfull" variant of the service is just the did.

> Now that "wrapping enum" is called ServiceResolvable , whether you call this struct FullService, or Service, it's equally distinct from ServiceResolvable and I don't see an ambiguity in either case.

My point was different - that the enum could be reasonably called `Service`.